cmake/cmakepp_lang/types/type.cmake

```diff
@@ -55,5 +56,10 @@ function(cpp_is_type _it_result _it_type)
         return()
     endif()
 
+    if ("${_it_type}" MATCHES ".+[*]")
```

Comment:
I think we should only be checking for pointers to a valid type. If the user provided `invalid*` right now, I think this would return that it is a valid type. I suggest detecting and stripping the trailing `*` first, then letting the usual type checking above test if the remainder is a valid type.

Reply:
Yeah that's what I meant by restricting the is_type check a bit, I can do that

Reply:
Seems your idea worked; thanks! Was simpler than I thought. Sorry, just finished an algorithms final yesterday so I'm a bit mentally exhausted and not thinking as clearly as I would like